Hymns/Tags/TagSubList.swift

```diff
@@ -0,0 +1,24 @@
+import SwiftUI
+import Resolver
+
+struct TagSubList: View {
+    let tagSelected: SongResultViewModel
+    @ObservedObject private var viewModel: TagListViewModel
+
+    init(viewModel: TagListViewModel = Resolver.resolve(), tagSelected: SongResultViewModel) {
+        self.viewModel = viewModel
+        self.tagSelected = tagSelected
+    }
+
+    var body: some View {
+        VStack {
+            List(self.viewModel.tags) { tag in
+                NavigationLink(destination: tag.destinationView) {
+                    SongResultView(viewModel: tag)
+                }
+            }
+        }.onAppear {
+            self.viewModel.fetchTagsByTags(self.tagSelected.title)
+        }
+    }
+}
```

Comment:
So I have this thing already called BrowseResultsListView and BrowseResultsListViewModel. I would like to reuse that as much as possible. I know right now it has code to fetch Category/Subcategory, but if we could make that more generic and use that, that'd be  preferable to creating this special sublist that basically just does the same thing

Reply:
oh I have an idea. What if we just overrode the init functions.

So you'd create your own init function:
```
    init(paramsThatYouNeed, dataStore: HymnDataStore = Resolver.resolve(),
         backgroundQueue: DispatchQueue = Resolver.resolve(name: "background"),
         mainQueue: DispatchQueue = Resolver.resolve(name: "main")) {

       // fill in the title and results

    }
```

Reply:
I'll try it. I've been looking at it the past hour or so. I was going a route of trying to use protocols and generics. But let me see if this solution is doable. Seems like it would be nice.

Reply:
I couldn't get that to work because the @ObservedObject still needs a type specified before the init function

Reply:
Can we just note this for a refactor later lol :/

Reply:
Are you referring to the protocol approach? I don't think that'll work. Did you try overriding the init function?

Reply:
Nice work!